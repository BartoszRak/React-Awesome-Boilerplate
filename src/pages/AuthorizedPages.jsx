import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'

import Home from './Home'

export function AuthorizedPages() {
  return (
    <Switch>
      <Route component={Home} path="/" />
      <Redirect from="*" to="/" />
    </Switch>
  )
}

export default AuthorizedPages
