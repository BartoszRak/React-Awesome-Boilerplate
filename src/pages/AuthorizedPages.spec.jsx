import React from 'react'
import { shallow } from 'enzyme'

import { AuthorizedPages } from './AuthorizedPages'

jest.mock('./Home', () => 'HomeMock')

const initComponent = overrides => {
  const mockProps = {}
  const mockMethods = {}
  const wrapper = shallow(<AuthorizedPages {...mockProps} {...mockMethods} {...overrides} />)
  return { mockProps, wrapper }
}

describe('AuthorizedPages: ', () => {
  it('renders without crashing', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toBeTruthy()
  })
  it('should render as expected', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toMatchSnapshot()
  })
})
