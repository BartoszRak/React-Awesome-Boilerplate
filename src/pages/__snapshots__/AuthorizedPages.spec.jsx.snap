// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorizedPages:  should render as expected 1`] = `
<SwitchMock>
  <RouteMock
    component="HomeMock"
    path="/"
  />
  <RedirectMock
    from="*"
    to="/"
  />
</SwitchMock>
`;
